pkg/controller/cabundleinjector/mutatingwebhook_test.go

```diff
@@ -0,0 +1,165 @@
+			clientObjects = append(clientObjects, mwc)
+		}
+	}
+	lister := admissionreglister.NewMutatingWebhookConfigurationLister(mutatingWebhookIndexer)
+	kubeclient := fake.NewSimpleClientset(clientObjects...)
+	m.injector = &mutatingWebhookCABundleInjector{
+		client:   kubeclient.AdmissionregistrationV1().MutatingWebhookConfigurations(),
+		lister:   lister,
+		caBundle: m.getCABundle(),
+	}
+	m.client = kubeclient.AdmissionregistrationV1().MutatingWebhookConfigurations()
+	m.syncContext = newTestSyncContext(mwc.Name)
+}
+
+func (m *mutatingInjectTest) sync() error {
+	return m.injector.Sync(context.TODO(), m.syncContext)
+}
+
+func (m *mutatingInjectTest) validateAllCABundles(t *testing.T, o runtime.Object) {
+	mwc, ok := o.(*admissionregv1.MutatingWebhookConfiguration)
+	if !ok {
+		t.Errorf("object is not of type MutatingWebhookConfiguration")
+	}
+
+	for _, webhook := range mwc.Webhooks {
+		if !bytes.Equal(webhook.ClientConfig.CABundle, m.getCABundle()) {
+			t.Errorf("expected %s, got %s", m.getCABundle(), webhook.ClientConfig.CABundle)
+		}
+	}
+}
+
+func TestMutatingWebhookCABundleInjectorSync(t *testing.T) {
+	m := mutatingInjectTest{}
+
+	// Common webhook config
+	webhookClientConfig := admissionregv1.WebhookClientConfig{
+		// A service must be specified for validation to
+		// accept a cabundle.
+		Service: &admissionregv1.ServiceReference{
+			Namespace: "foo",
+			Name:      "foo",
+		},
+	}
+	sideEffectNone := admissionregv1.SideEffectClassNone
+
+	mwc := &admissionregv1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: m.getWebhookName(),
+			Annotations: map[string]string{
+				api.InjectCABundleAnnotationName: "true",
+			},
+		},
+		Webhooks: []admissionregv1.MutatingWebhook{
+			// Specify 2 webhooks to ensure more than 1 webhook will be updated
+			{
+				Name:                    "ut-1.example.com",
+				ClientConfig:            webhookClientConfig,
+				SideEffects:             &sideEffectNone,
+				AdmissionReviewVersions: []string{"v1"},
+			},
+			{
+				Name:                    "ut-2.example.com",
+				ClientConfig:            webhookClientConfig,
+				SideEffects:             &sideEffectNone,
+				AdmissionReviewVersions: []string{"v1"},
+			},
+		},
+	}
+
+	m.injectorSetup(t, mwc)
+	if _, err := m.client.Create(context.TODO(), mwc, metav1.CreateOptions{}); err != nil {
+		t.Error(err)
+	}
+
+	if err := m.sync(); err != nil {
+		t.Error(err)
+	}
+
+	mwc, err := m.client.Get(context.TODO(), m.getWebhookName(), metav1.GetOptions{})
+	if err != nil {
+		t.Error(err)
+	}
+	m.validateAllCABundles(t, mwc)
+	mwc.Webhooks = append(mwc.Webhooks, admissionregv1.MutatingWebhook{
+		Name:                    "ut-3.example.com",
+		ClientConfig:            webhookClientConfig,
+		SideEffects:             &sideEffectNone,
+		AdmissionReviewVersions: []string{"v1"},
+	})
+
+	mwc, err = m.client.Update(context.TODO(), mwc, metav1.UpdateOptions{})
+	if err != nil {
+		t.Error(err)
+	}
+
+	m.injectorSetup(t, mwc)
```

Comment:
@stlaz I wasn't sure how to get the update to take effect without recreating the injector. I followed `secret_creating_controller_test.go` to get an idea of how to setup the kube client. If any better direction here, please let me know. Thanks.